client/SFTPClientGUI/SFTPClientGUI.py

```diff
@@ -0,0 +1,351 @@
+				self._tvmodel.append(current, [None, None, None, None, None, None, None])
+			counter += 1
+		if counter < 1:
+			self._tvmodel.append(parent, [None, None, None, None, None, None, None])
+
+	def delete(self, model, treeiter):
+		try:
+			if model[treeiter][5] == -1:
+				shutil.rmtree(model[treeiter][2])
+			else:
+				os.remove(model[treeiter][2])
+			self.logger.log_generic('Successfully deleted ' + model[treeiter][2])
+			self.refresh()
+		except OSError:
+			self.logger.log_generic("Permissions Error Deleting File")
+
+class RemoteDirectory(DirectoryBase):
+	def __init__(self, treeview, logger, download_button, application, ftp, *args, **kwargs):
+		super(RemoteDirectory, self).__init__(treeview, logger, *args, **kwargs)
+		self.ftp = ftp
+		self.download_button = download_button
+		self.application = application
+		self.def_dir = self.application.config['server_config']['server.web_root']
+		self.load_dirs(self.def_dir)
+		self.treeview_local.set_model(self._tvmodel)
+
+	def load_dirs(self, path, parent=None):
+		self.ftp.chdir(path)
+		counter = 0
+		for name in self.ftp.listdir(path):
+			if self.local_hidden and name.startswith('.'):
+				continue
+			fullname = path + '/' + name
+			lstatout = self.ftp.lstat(fullname)
+			perm = '   '
+			_file = self.ftp.file(fullname)
+			perm += 'r' if _file.readable() else '-'
+			perm += 'w' if _file.writable() else '-'
+			perm += '-'
+			_file.close()
+			date_modified = '   ' + time.strftime("%y-%m-%d %H:%M:%S", time.localtime(lstatout.st_mtime))
+			isFolder = stat.S_ISDIR(lstatout.st_mode)
+			if isFolder:
+				icon = Gtk.IconTheme.get_default().load_icon('folder', 20, 0)
+				current = self._tvmodel.append(parent, (name, icon, fullname, perm, None, -1, date_modified))
+			else:
+				file_size = lstatout.st_size
+				hr_file_size = '   ' + boltons.strutils.bytes2human(file_size)
+				icon = Gtk.IconTheme.get_default().load_icon('empty', 12.5, 0)
+				current = self._tvmodel.append(parent, (name, icon, fullname, perm, hr_file_size, file_size, date_modified))
+			if isFolder:
+				self._tvmodel.append(current, [None, None, None, None, None, None, None])
+			counter += 1
+		if counter < 1:
+			self._tvmodel.append(parent, [None, None, None, None, None, None, None])
+
+	def delete(self, model, treeiter):
+		try:
+			self.ftp.remove(model[treeiter][2])
+			self.logger.log_generic('Successfully deleted ' + model[treeiter][2])
+			self.refresh()
+		except IOError:
+			self.logger.log_generic("Permissions Error Deleting File")
+
+class FileManager(object):
+	def __init__(self, builder, application, ftp, *args, **kwargs):
+		self.builder = builder
+		self.application = application
+		upload_button = self.builder.get_object('button_upload')
+		download_button = self.builder.get_object('button_download')
+		self.logger = Logger(self.builder)
+		self.local = LocalDirectory(self.builder.get_object('SFTPClientGUI.treeview_local'), self.logger, upload_button, ftp)  # pylint: disable=unused-variable
```

Comment:
The builder itself should be passed to the (Local/Remote)Directory classes and ultimately to `DirectoryBase` which should load the treeview and set an instance attribute based on the name (i.e. `treeview_local` for LocalDirectory) in a class attribute.

Something like:
```python
class DirectoryBase(object):
    def __init__(self.builder):
        self._treeview = self.builder.get_object('SFTPClientGUI.' + self.treeview_name)
        ...

class LocalDirectory(DirectoryBase):
    treeview_name = 'treeview_local'
    def __init__(self, *args, **kwargs):
        super(LocalDirectory, self).__init__(*args, **kwargs)
        ...
```